app/components/display/search_result_component.rb

```diff
@@ -34,6 +34,21 @@ def details_button
       end
   end
 
+  def mappings_button
+    link_to_modal(nil, "/ajax/mappings/get_concept_table?ontologyid=#{@ontology_acronym}&conceptid=#{CGI.escape(@uri)}&type=modal", data: { show_modal_title_value: @title, show_modal_size_value: 'modal-xl' }) do
+      content_tag(:div, class: 'button') do
+        inline_svg_tag('icons/ontology.svg') +
+        content_tag(:div, class: 'text d-flex') do
+          render(TurboFrameComponent.new(id: 'mapping_count', src: "/ajax/mappings/get_concept_table?ontologyid=#{@ontology_acronym}&conceptid=#{CGI.escape(@uri)}", loading: "lazy")) do |t|
```

Comment:
don't use `CGI.escape(@uri)`, just use `escape(@uri)` (include the helper file if needed)

Reply:
I didn't want to include the application helper file here which is a very big file

Reply:
create a file, called `url_helper.rb`, where you put the `escape` and `unescape` helpers.
then include it in the `ApplicationHelper`, and in this component.